op/fdg_watcher.py

```diff
@@ -109,42 +171,86 @@ def main():
     parser.add_argument("--to_addr", type=str, default="", help="email to address")
     parser.add_argument("--username", type=str, default="", help="email username")
     parser.add_argument("--password", type=str, default="", help="email password")
-    parser.add_argument("--test_email", type=bool, default=False, help="send a test email when start")
-
+    parser.add_argument("--test_email", action="store_true", help="send a test email when start")
+    
     args = parser.parse_args()
 
     # prev_balance = None
     prev_send = time.monotonic()
 
     if args.test_email:
-        title = "test email for {}".format(args.recipient)
+        title = "test email for fdg watcher {}".format(args.fdg_factory)
         send_email(title, "", args.from_addr, args.to_addr, args.username, args.password)
 
-    bn = get_blocknumber(args.l1_rpc)
-    games = get_fdg_games(args.l1_rpc, args.fdg_factory, bn - args.blocks)
+    try:
+        while True:
+            logger.info("Checking")
+            errors = 0
+            blacklisted_count = 0  # Counter for blacklisted games
+            msg = ""
+
+            try:
+                bn = get_blocknumber(args.l1_rpc)
+                games = get_fdg_games(args.l1_rpc, args.fdg_factory, bn - args.blocks)
+            except Exception as e:
+                logger.error(f"Failed to get block number or games: {e}")
+                time.sleep(60)  # Wait a minute before retrying
+                continue
 
-    while True:
-        logger.info("Checking")
-        errors = 0
-        msg = ""
-        for g in games:
-            info = get_fdg_game_info(args.l1_rpc, g["transactionHash"])
-            l2output = get_l2output(args.l2_rpc, info["blockNumber"])
-            if l2output != "0x" + info["output"]:
-                msg += "error: game {}, expected {}, actual {}\n".format(g["transactionHash"], l2output,info["output"])
-                errors += 1
+            for g in games:
+                info = get_fdg_game_info(args.l1_rpc, g["transactionHash"])
+                # Skip if blockNumber is 0 (indicating an error in getting game info)
+                if info["blockNumber"] == 0 or not info["output"]:
+                    logger.warning(f"Skipping game check due to missing data: {g['transactionHash']}")
```

Comment:
This may silently skip problematic games without alerting, maybe track and report skipped games in email alerts.